oak-run-commons/src/main/java/org/apache/jackrabbit/oak/indexversion/IndexVersionOperation.java

```diff
@@ -254,6 +242,24 @@ enum Operation {
         DELETE_HIDDEN_AND_DISABLE,
         DELETE
     }
+
+    protected abstract IndexVersionOperation getIndexVersionOperationInstance(IndexName indexName);
+
+    /**
+     *
+     * @param indexNode - NodeState of a disabled index
+     * @return true if the disabled index with NodeState indexNode can be marked for deletion or not.
+     */
+    protected abstract boolean checkIfDisabledIndexCanBeMarkedForDeletion(NodeState indexNode);
+
+    /**
+     *
+     * @param reverseSortedIndexNameList
+     * @param parentPath
+     * @param rootNode
+     * @return Highest versioned active index's IndexName
+     */
+    protected abstract IndexName getActiveIndex(List<IndexName> reverseSortedIndexNameList, String parentPath, NodeState rootNode);
```

Comment:
This method should be annotated with `@Nullable` since it can return a null value.

Reply:
ack.